edfio/edf.py

```diff
@@ -698,6 +707,11 @@ def _set_num_data_records_with_signals(
                 signal_durations[0],
                 self.data_record_duration,
             )
+            signal_lengths = [len(s._digital) for s in signals]
+            if any(l % num_data_records for l in signal_lengths):
```

Comment:
We could change this logic such that it also considers the sampling frequency of the signals. I don't think it's sufficient to verify that the total number of sample divides evenly by the number of records.

Reply:
I was not able to come up with a test that would show this issue... maybe using each sampling frequency to calculate signal durations, which are then used to  determine `num_data_records` already covers that?